project_1/src/simulator.cpp

```diff
@@ -2,60 +2,86 @@
 
 NetworkSimulator::NetworkSimulator(const std::vector<std::uint32_t>& distances_to_receiver) {
     for (std::uint32_t distance: distances_to_receiver) {
-        servers.emplace_back(GenerateNewID(servers), distance);
+        servers.emplace_back(GenerateNewID(servers), distance, packets);
     }
     std::uint32_t server_receiver_id = 0;
     server_receiver = ServerBase(server_receiver_id);
 }
 
-void NetworkSimulator::StartSimulation() {
+void NetworkSimulator::PacketsSending(bool is_first_iteration) {
```

Comment:
I agree that good code must avoid flags like this one, but in current implementation of packet sending this flag is necessary. If we remove it, servers will never send packets (because on every iteration `continue` will be executed).

I will try to resolve this code-design problem.